free-1.md

```diff
@@ -9,42 +9,141 @@ good way to meet their needs.
+okay with allowing me to take notes on the computer in any class. I found
+that social sciences classes which require memorization, teachers
+who did not necessarily write every important definition or point out but rather
+preferred to talk about it, could be extremely frustrating if good notes weren't
+taken. I became a great note taker with my computer and, though studies show
+otherwise, felt like I absorbed more through taking robust, detailed notes on 
+the computer rather than listening and hand-writing notes sparingly. The biggest
+issue came in college economics, my major, when many graphs came into play and
+therefore made computer notes very difficult.  Using Evernote, clipping pictures
+of the relevant graphs from Google images was the only solution, but was too
+inefficient to continue doing.  So, I thought having a domain specific language
+for this would be a great solution. 
 
 ### Domain
 _Describe the project's domain in five words._
-
+Creating graphs for economics notes
 
 ### Interface (syntax)
 _How might the user interact with the language? What does programming look 
 like? Why is this the right way to interact with the problem domain?_ 
-
+The user would likely be using some sort of text editor or word processor for
+taking notes. This is an early problem for me: can I create the language to be
+usable in popular word processors? Regardless, I would imagine some note snippets
+to look similar to the following in this language:
+
+``` 
+This is a standard supply and demand graph
+
+Graph{
+	Y-Axis: Price
+	X-Axis: Quantity
+	SupplyCurve()	
+	DemandCurve()
+}
+
+generate Graph
+```
+
+```
+Today, we're learning about the *Engel Curve*, the graph looks like this: 
+
+Graph{
+	Y-axis: Income/month
+	X-axis: Units
+	EngelCurve()
+}
+
+generate Graph
+```	
+
+By instantiating some Graph object, you can generate a graph in place in the 
+notes. Then the user could label axes and specify the curves that are needed 
+in the graph. So you could specify SupplyCurve(), which would give a default
+upward sloping supply curve. Each curve could take certain parameters for
+transformations, for example `SupplyCurve(Slope: .5, Y: -10, X: 50  )`, would
+give you a supply curve with slope of .5 starting at a Y-value of -10 and X-value
+of 50. Also, we'd want to support markdown in the language to give users
+easy formatting options when writing notes.
 
 ### Operation (semantics)
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
+The program would ideally run on save.  When it runs, the language would generate
+a graph image and then copy it to the clipboard for the user to paste or
+ideally make it appear in place, but this might be difficult. An early issue I 
+could forsee happening is if a user specifies a curve that is not supported in
+the program.  For example, if they specified `ProductionFunction()` and we didn't
```

Comment:
How are you going to design for extensibility? Can users create their own functions?